### PR TITLE
[SPARK-17328][SQL] Fix NPE with EXPLAIN DESCRIBE TABLE

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This PR fixes the following NPE scenario in two ways.

**Reported Error Scenario**

``` scala
scala> sql("EXPLAIN DESCRIBE TABLE x").show(truncate = false)
INFO SparkSqlParser: Parsing command: EXPLAIN DESCRIBE TABLE x
java.lang.NullPointerException
```
- **DESCRIBE**: Extend `DESCRIBE` syntax to accept `TABLE`.
- **EXPLAIN**: Prevent NPE in case of the parsing failure of target statement, e.g., `EXPLAIN DESCRIBE TABLES x`.
## How was this patch tested?

Pass the Jenkins test with a new test case.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala [hunk 1]
```diff
@@ -265,7 +265,9 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
     }
 
     val statement = plan(ctx.statement)
-    if (isExplainableStatement(statement)) {
+    if (statement == null) {
+      null  // This is enough since ParseException will raise later.
```
Comment (hvanhovell) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: Where will the parseException be raised? Could you be a bit more specific in your comment?

Maybe add a small test for this? Could you also add a few unit tests (not end-2-end) to SparkSqlParserSuite.

**Review by hvanhovell — CHANGES_REQUESTED** [review]
Looks pretty good. Left one test related comment.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala (hunk 1 above)
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: Sure!

Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: I'll add a testcase first~

Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: I added the testcase to intercept that. If it returns `null`, `ParseDriver.scala` recognizes it as a parsing error and raises `Unsupported SQL statement`.
```
  override def parsePlan(sqlText: String): LogicalPlan = parse(sqlText) { parser =>
    astBuilder.visitSingleStatement(parser.singleStatement()) match {
      case plan: LogicalPlan => plan
      case _ =>
        val position = Origin(None, None)
        throw new ParseException(Option(sqlText), "Unsupported SQL statement", position, position)
    }
  }
```

**Comment by hvanhovell** [comment]
LGTM - merging to master

**Comment by dongjoon-hyun** [comment]
Thank you, @hvanhovell !

**Pull request closed without merging**